CONTRIBUTING.md

```diff
@@ -100,6 +101,50 @@ Be sure to include any related GitHub issue references in the commit message.  S
 [GFM syntax](https://guides.github.com/features/mastering-markdown/#GitHub-flavored-markdown) for referencing issues
 and commits.
 
+
+[dronevic]:https://ci.vmware.run/vmware/vic
+[e2edronevic]:https://e2e.ci.vmware.run/vmware/vic
+[dronesrc]:https://github.com/drone/drone
+[dronecli]:http://readme.drone.io/devs/cli/
+
+## Automated Testing
+
+Automated testing uses [Drone][dronesrc].
+
+PRs must pass unit tests and integration tests before being merged into `master`.
+
+You can run the tests locally before making a PR or view the Drone build results for [unit tests][dronevic]
+and [integration tests][e2edronevic].
+
+If you don't have a running ESX required for tests, you can leverage the automated Drone servers for
+running tests. Add `WIP` (work in progress) to the PR title to alert reviewers that the PR is not ready to be merged.
+
+
+### Testing locally
+
+Developers need to install [Drone CLI][dronecli].
+
+#### Unit tests
+
+``` shell
+drone exec --yaml .drone.yml -e VIC_ESX_TEST_URL="<USER>:<PASS>@<ESX IP>"
+```
+
+If you don't have a running ESX, tests requiring an ESX can be skipped with the following:
+
+``` shell
+drone exec --yaml .drone.yml -e VIC_ESX_TEST_URL=""
+```
+
+#### Integration tests
+
+Integration tests require a running ESX on which to deploy VIC.
+
+``` shell
+drone exec --yaml .drone-e2e.yml -e VIC_ESX_TEST_URL="<USER>:<PASS>@<ESX IP>"
```

Comment:
It is missing temporarily.  I can clean this up if you want, but as soon as we fix the infra problems we are having we are going to add it back in and start using it again.

Mikkel Hagen
Quality Engineering Lead
Phone: (716) 307 0362
Address: 6500 River Pl Blvd, Austin, TX 78730

On 05/28/2016 07:27 PM, Faiyaz Ahmed wrote:

In CONTRIBUTING.md<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_vmware_vic_pull_707-23discussion-5Fr64996062&d=CwMCaQ&c=Sqcl0Ez6M0X8aeM67LKIiDJAXVeAw-YihVMNtXt-uEs&r=RWOEbpwLdzFOxVxJx1-lNmFIV0bt5SZaCb_OGY7aD4c&m=-q6FoYWibSNZ6I7DNEtnk60Tlh1sn5ttZ3kl19_rl3w&s=zW1CihO43cT32Kn4JbH821eBX1BO-RyfgAuqcXWtXzs&e=>:

> +``` shell
> +drone exec --yaml .drone.yml -e VIC_ESX_TEST_URL="<USER>:<PASS>@<ESX IP>"
> +```
> +
> +If you don't have a running ESX, tests requiring an ESX can be skipped with the following:
> +
> +``` shell
> +drone exec --yaml .drone.yml -e VIC_ESX_TEST_URL=""
> +```
> +
> +#### Integration tests
> +
> +Integration tests require a running ESX on which to deploy VIC.
> +
> +``` shell
> +drone exec --yaml .drone-e2e.yml -e VIC_ESX_TEST_URL="<USER>:<PASS>@<ESX IP>"


This file (.drone-e2e.yml) looks missing from this changeset. Or am I just not seeing it?

-
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_vmware_vic_pull_707_files_d3c7fb8e6e6916ae2d1accbcaad7e9384f649be5-23r64996062&d=CwMCaQ&c=Sqcl0Ez6M0X8aeM67LKIiDJAXVeAw-YihVMNtXt-uEs&r=RWOEbpwLdzFOxVxJx1-lNmFIV0bt5SZaCb_OGY7aD4c&m=-q6FoYWibSNZ6I7DNEtnk60Tlh1sn5ttZ3kl19_rl3w&s=7R8Agdu_yCLxhHRNI0q3SKkydKzgQVUs36iY7voznxM&e=>, or mute the thread<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe_ASfq-2DXL3abElyFWuy-5FQCE6v73ouUhoCFks5qGN1SgaJpZM4Igvq6&d=CwMCaQ&c=Sqcl0Ez6M0X8aeM67LKIiDJAXVeAw-YihVMNtXt-uEs&r=RWOEbpwLdzFOxVxJx1-lNmFIV0bt5SZaCb_OGY7aD4c&m=-q6FoYWibSNZ6I7DNEtnk60Tlh1sn5ttZ3kl19_rl3w&s=seogGPvUNHaIRa8gUT9jbPakA6n3v3CB0BmFg5AK4kU&e=>.